Hazel/src/Hazel/Core/Core.h

```diff
@@ -44,12 +44,17 @@
 #endif // End of platform detection
 
 #ifdef HZ_DEBUG
+	#ifdef HZ_PLATFORM_WINDOWS
+		#define HZ_DEBUGBREAK() __debugbreak()
+	#else
+		#error "Platform doesn't support debugbreak yet!"
+	#endif
```

Comment:
I would already implement the linux equivalent since quite a few are on linux, and it is not the only macro that is already implemented for linux.

```suggestion
 	#if defined HZ_PLATFORM_WINDOWS 
 		#define HZ_DEBUGBREAK() __debugbreak() 
 	#elif defined HZ_PLATFORM_LINUX 
 		#include <signal.h> 
 		#define HZ_DEBUGBREAK() raise(SIGTRAP)
	#else
		#error "Platform doesn't support debugbreak yet!"
	#endif
```

Reply:
The only concern I might have is the inconsistency with regards to brackets as used here:
https://github.com/TheCherno/Hazel/blob/eb3f705b6097d81d8c743d19ecb239d4dfb94c0c/Hazel/src/Hazel/Core/Core.h#L35-L44

Reply:
if we can, we leave the brackets out of the macro. However for debugbreak, on linux you need an argument between the brackets, so it's not something we can leave out:

```
raise(SIGTRAP)(); // Error when just removing the brackets
raise(); // If brackets are not part of it, it won't break correct, missing argument
```

Also, similar to the asserts, there the brackets are also part of the macro as it has an argument. A check never has brackets, and for function macros, the rule is to always add brackets to the call, but in the macro definition, the brackets can be omitted. So the definition can look inconsistent, in the code base, where it is called, it is always consistent with the brackets always present.

Reply:
I meant something like:
```suggestion
	#if defined(HZ_PLATFORM_WINDOWS)
		#define HZ_DEBUGBREAK() __debugbreak()
	#elif defined(HZ_PLATFORM_LINUX)
		#include <signal.h>
		#define HZ_DEBUGBREAK() raise(SIGTRAP)
	#else
		#error "Platform doesn't support debugbreak yet!"
	#endif
```

Reply:
oh yes, that's perfectly fine